packages/Search/test/tstLinearBVH.cpp

```diff
@@ -190,26 +190,19 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, empty_tree, DeviceType )
                   } ),
                   {}, {0, 0, 0}, success, out );
 
-    // NOTE: Expected output for nearest queries when asking for more neighbors
-    // than there are leaves in the tree is not ideal but that is what the
-    // current implementation does to avoid two passes in the search (count
-    // first, then allocate and save results in 2nd pass) `-1` denotes invalid
-    // index and `infty` invalid distance.
     checkResults( empty_bvh,
                   makeNearestQueries<DeviceType>( {
                       {{{0., 0., 0.}}, 1},
                       {{{1., 1., 1.}}, 2},
                   } ),
-                  {-1, -1, -1}, {0, 1, 3}, success, out );
+                  {}, {0, 0, 0}, success, out );
 
-    double const infty = std::numeric_limits<double>::max();
     checkResults( empty_bvh,
                   makeNearestQueries<DeviceType>( {
                       {{{0., 0., 0.}}, 1},
                       {{{1., 1., 1.}}, 2},
                   } ),
-                  {-1, -1, -1}, {0, 1, 3}, {infty, infty, infty}, success,
-                  out );
+                  {}, {0, 0, 0}, {}, success, out );
```

Comment:
This change summarizes well the changes introduced in that PR.

Reply:
BEFORE: would return invalid indices/distances when unable to find neighbors.
AFTER: only return what was actually found.